src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6054,7 +6053,7 @@ private BoundExpression BindArrayAccess(ExpressionSyntax node, BoundExpression e
                 }
             }
 
-            return new BoundArrayAccess(node, expr, convertedArguments.AsImmutableOrNull(), arrayType.ElementType);
+            return new BoundArrayAccess(node, expr, convertedArguments.AsImmutableOrNull(), arrayType.ElementType, hasErrors);
```

Comment:
We aren't setting `hasErrors` for errors converting array index arguments above. Is there a reason to set `hasErrors` for `ref` arguments errors only?